src/mds/Server.cc

```diff
@@ -3435,12 +3436,21 @@ void Server::handle_client_readdir(MDRequestRef& mdr)
     mdcache->lru.lru_touch(dn);
   }
   
-  __u8 complete = (end && offset_str.empty());  // FIXME: what purpose does this serve
+  bool complete = false;
+  __u16 flags = 0;
+  if (end) {
+    flags = CEPH_READDIR_FRAG_END;
+    complete = offset_str.empty(); // FIXME: what purpose does this serve
```

Comment:
I realize you didn't introduce this comment, but let's try and figure it out. It looks to me as though this exists from 60ec1cbe0ecc1e57d22c0eccffcb3b30faf7f5b9, and was never referenced in the userspace Client at all. Does the kernel look at it?